qa/tests/20-tunnel-readiness-perf.sh

```
@@ -0,0 +1,157 @@
+  echo "      PI_QA_TUNNEL_SOAK=1. Skipped loudly on purpose: a soak that silently"
+  echo "      passes because nothing was running is worse than no soak."
+  exit 0
+fi
+
+echo "=== Test: concurrent-tunnel soak (P3) ==="
+SOAK_MIN=${PI_QA_SOAK_MINUTES:-30}
+RSS_GROWTH_PCT=10
+
+pid_of_server() { curl -fsS "${BASE}/api/health" | sed -n 's/.*"pid":\([0-9]*\).*/\1/p'; }
+rss_kb() { ps -o rss= -p "$1" 2>/dev/null | tr -d ' '; }
+child_pids() { ls "${HOME}/.pi/dashboard"/*.pid 2>/dev/null | wc -l | tr -d ' '; }
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)